pkg/cvo/cvo.go

```diff
@@ -223,17 +227,32 @@ func New(
 // controller that loads and applies content to the cluster. It returns an error if the payload appears to
 // be in error rather than continuing.
 func (optr *Operator) InitializeFromPayload(restConfig *rest.Config, burstRestConfig *rest.Config) error {
-	update, err := payload.LoadUpdate(optr.defaultPayloadDir(), optr.releaseImage, optr.exclude)
+	update, err := payload.LoadUpdate(optr.defaultPayloadDir(), optr.release.Image, optr.exclude)
 	if err != nil {
 		return fmt.Errorf("the local release contents are invalid - no current version can be determined from disk: %v", err)
 	}
-	// XXX: set this to the cincinnati version in preference
-	if _, err := semver.Parse(update.ImageRef.Name); err != nil {
-		return fmt.Errorf("the local release contents name %q is not a valid semantic version - no current version will be reported: %v", update.ImageRef.Name, err)
+	optr.release.Version = update.Metadata.Version
+	if urlInterface, ok := update.Metadata.Metadata["url"]; ok {
+		if urlString, ok := urlInterface.(string); ok {
+			optr.release.URL = configv1.URL(urlString)
+		} else {
+			klog.Warningf("the local release contents define a non-string URL: %v", urlInterface)
+		}
+	}
+	if channelsInterface, ok := update.Metadata.Metadata["io.openshift.upgrades.graph.release.channels"]; ok {
+		if channelsString, ok := channelsInterface.(string); ok {
+			optr.release.Channels = strings.Split(channelsString, ",")
+			sort.Strings(optr.release.Channels)
+		} else {
+			klog.Warningf("the local release contents define a non-string URL: %v", channelsInterface)
+		}
+	}
+
```

Comment:
Shouldn't `optr.release.Version = update.Metadata.Version` be moved here

Reply:
> Shouldn't optr.release.Version = update.Metadata.Version be moved here

Why?  It's a few lines up, and `InitializeFromPayload` shouldn't be racing serious activity in other goroutines (because what would those routines be doing before we'd initialized ourselves?).

Reply:
Nothing to do with a race, just keeping it close to were it's being checked/used for first time which is typically a best-practice. But not anything critical so I'm okay with doing nothing if you prefer.